firebase_functions/functions/src/helpers.ts

```diff
@@ -163,11 +163,72 @@ const faucet = async receiveAddress => {
   );
 };
 
+
+const getBalance = async address => {
+  try {
+    if (!address) {
+      return 0;
+    }
+    const { provider } = await getSettings();
+    const { getBalances } = composeAPI({ provider });
+    const { balances } = await getBalances([address], 100);
+    return balances && balances.length > 0 ? balances[0] : 0;
+  } catch (error) {
+    console.error('getBalance error', error);
+    return 0;
+  }
+};
+
+
+const repairWallet = async (seed, keyIndex) => {
+  try {
+    const address = await generateAddress(seed, keyIndex)
+    const balance = await getBalance(address)
+  
+    if (balance === 0){
+    const iterable = [-2, -1, 0, 1, 2, 3, 4, -3, -4,-5,-6-7,5,6,7];
+
+    for (let value of iterable) {
+
+      if ((value < 0 && Math.abs(value) <= keyIndex) || value >= 0) {
+        const newIndex = Number(keyIndex) + Number(value)
+        const newAddress = await generateAddress(seed, newIndex)
+        const newBalance = await getBalance(newAddress);
+
+        if (newBalance > 0) {
+          return { address: newAddress, keyIndex: newIndex, seed };
+        }
+      }
+      value += 1;
```

Comment:
you are iterating over `iterable` and assign it to `value`. 
why do you need to increase the value at the end of the iteration?